Add documentation for GCP Service Classes (#1717)
kyma-project/kyma: docs/service-brokers/gcp-broker-service-classes/cloud-sql-mysql/docs/plans-details.md

| **authorizedGaeApplications** | `string` | The App Engine application IDs that can access this instance. This property is only applicable to First Generation instances. | NO | - |
| **backupConfiguration** | `string` | The daily backup configuration for the instance. For more information, go to the **BackupConfiguration properties** section. | NO | - |
| **crashSafeReplicationEnabled** | `boolean` | Configuration specific to read replica instances. Indicates whether database flags for crash-safe replication are enabled. This property is only applicable to First Generation instances. | NO | - |
| **dataDiskSizeGb** | `string` | The size of data disk in `GB`. The data disk size minimum is `10GB`. Applies only to Second Generation instances. | NO | - |
| **dataDiskType** | `string` | The type of data disk. The possible values are `PD_SSD`, `PD_HDD`. Applies only to Second Generation instances. | NO | `PD_SSD` |
| **databaseFlags** | `array` | The database flags passed to the instance at startup. | NO | - |

> | **databaseFlags.name** | `string` | The name of the flag. These flags are passed at instance startup, so include both MySQL server options and MySQL system variables. Flags should be specified with underscores, not hyphens. For more information, see **Configuring MySQL Flags** in the Google Cloud SQL documentation, as well as the official MySQL documentation for server options and system variables. | NO | - |

Reviewer: Wondering if there is a link to google documentation that we can put here? or it's not recommended :) To point to
`**Configuring MySQL Flags** in the Google Cloud SQL documentation`

Author: I've deleted the reference.

| **ipConfiguration** | `object` | The settings for IP Management. This allows to enable or disable the instance IP and manage which external networks can connect to the instance. The IPv4 address cannot be disabled for Second Generation instances. For more information, go to the **IpConfiguration properties** section. | NO | - |
| **locationPreference** | `object` | The location preference settings. This allows the instance to be located as near as possible to either an App Engine application or Compute Engine zone for better performance. App Engine co-location is only applicable to First Generation instances. | NO | - |
| **locationPreference.followGaeApplication** | `string` | The AppEngine application to follow. It must be in the same region as the Cloud SQL instance. | NO | - |
| **locationPreference.zone** | `string` | The preferred Compute Engine zone. | NO | - |
| **maintenanceWindow** | `object` | The maintenance window for this instance. This specifies when the instance may be restarted for maintenance purposes. Applies only to Second Generation instances. | NO | - |
| **maintenanceWindow.day** | `integer` | Day of week (1-7), starting on Monday. | NO | - |

> | **maintenanceWindow.hour** | `integer` | The hour of day  (0-23). | NO | - |

Reviewer: ```suggestion
| **maintenanceWindow.hour** | `integer` | The hour of the day  (0-23). | NO | - |
```

|----------------|------|-------------|----------|---------------|
| **enabled** | `boolean` | Indicates if this configuration is enabled. | NO | - |
| **binaryLogEnabled** | `boolean` | Indicates if the binary log is enabled. If backup configuration is disabled, binary log must be disabled as well. | NO | - |
| **startTime** | `string` | Start time for the daily backup configuration in UTC timezone in the 24 hour, `HH:MM`, format. | NO | - |

### IpConfiguration properties

These are the **IpConfiguration** properties:

| Parameter Name | Type | Description | Required | Default Value |
|----------------|------|-------------|----------|---------------|
| **authorizedNetworks** | `array` | The list of external networks that are allowed to connect to the instance using the IP. In CIDR notation, also known as slash notation. | NO | - |
| **authorizedNetworks.expirationTime** | `string` | The time in the RFC 3339 format when this access control entry expires. | NO | - |
| **authorizedNetworks.name** | `string` | An optional label to identify this entry. | NO | - |
| **authorizedNetworks.value** | `string` | The whitelisted value for the access control list. | NO | - |

> | **ipv4Enabled** | `boolean` | Whether the instance should be assigned an IP address or not. | NO | - |

Reviewer: ```suggestion
| **ipv4Enabled** | `boolean` | Indicates if the instance should be assigned an IP address or not. | NO | - |
```

> | **requireSsl** | `boolean` | Indicates whether SSL connections over IP should be enforced or not. | NO | - |

Reviewer: ```suggestion
| **requireSsl** | `boolean` | Indicates if SSL connections over IP should be enforced or not. | NO | - |
```
